podvm/files/etc/agent-config.toml

```diff
@@ -5,6 +5,10 @@ enable_signature_verification=false
 # When using the agent-config.toml the KATA_AGENT_SERVER_ADDR env var seems to be ignored, so set it here
 server_addr="unix:///run/kata-containers/agent.sock"
 
+# This field sets up the KBC that attestation agent uses
+# This is replaced in the makefile steps so do not set it manually
+aa_kbc_params = ""
```

Comment:
Is it reasonable to use `offline_fs_kbc: null` as default value here as this is the most common use case today? We can replace it via real settings from makefile and podvm build scripts.

Reply:
Hey QiFeng, so I think the problem with this is that we aren't (currently) always building the attestation-agent into the peer pod vm, so if we default `aa_kbc_params = "offline_fs_kbc: null"` in the agent-config.toml then the kata-agent would try and launch the binary which might not exist and then fail. That's why I suggested leaving it blank for now, but I'm open to other suggestions.

Reply:
Thanks Steve, I found the algorithm here https://github.com/kata-containers/kata-containers/blob/CCv0/src/agent/src/image_rpc.rs#L101, I'm a little confused, why don't we use existing of `AA_PATH` as a condiftion but to combine `AA_PATH` and `aa_kbc_params` when launch attestation-agent? Is there a case both skopeo and attestation-agent binaries provided in the guest?

Reply:
Sorry to reply to myself, but I think the attestation-agent launch decision might be more interesting than I covered above, and I missed some thing, so my brain has been turning over the problem more, and it's worth me dumping this to 'paper' before making any decisions:

*Only check `aa_kbc_params` isn't empty*
This is the current case. I could lead to a scenario (particularly if we ever want to use image offload for non-confidential cases such as peer pods avoid nested virtualisation issues, or people first playing with CoCo with no attestation set) that the `aa_kbc_params` is configured, but the attestation-agent binary is not in the guest image.
This would result in an error on image pull (that may or may not be understandable to a user). There is an argument that this isn't great, but also one that says if a user thinks they are using an attestation-agent-kbc but aren't then we'd want to fail.

*Just check the presence of the attestation-agent in the guest image*
This breaks down into a couple of 'mis-configured' use cases:
- *The attestation-agent binary is in the guest image, but the `aa_kbc_param` hasn't been configured* - I don't think this will currently break anything as we've added check for `aa_kbc_param` before using the attestation-agent features, but in future this could be a trap that a dev falls into and tries to use the `attestation_agent_started` setting for something. I'm also not sure if this option could confuse the image-rs logic.
- *The attestation-agent binary isn't in the guest image, but the `aa_kbc_param` has been configured* - I think this is the worse case as a user has configured to use the a kbc, but the attestation-agent won't be started. Best case scenario the user thinks they have more security than they do, worst case (and I think the current situation with the code base) is things won't work as they will be configured to use attestation-agent config, but the binary isn't there

*Check the presence of the attestation-agent in the guest image and for offline_fs_kbc being set*
This breaks down into the following use cases:
- *The attestation-agent binary is in the guest image, and the `aa_kbc_param` is configured* - This is the good case, so we'd continue with no issues
- *The attestation-agent binary is in the guest image, but the `aa_kbc_param` hasn't been configured* - We'd not start the attestation-agent here, which I think is also a fine decision
- *The attestation-agent binary isn't in the guest image, but the `aa_kbc_param` has been configured* - This is the interesting case to consider here. Under this case we wouldn't start the attestation-agent, which would protect against the problem we would have in the current code base where the call to start the attestation-agent binary fails here, but there is a question of what the correct behaviour should be under this scenario. Do we want to proceed with no attestation-agent running (and end up with the later code that reads for config `aa_kbc_param` failing, should we add extra protection to those uses of `aa_kbc_param` to check the attestation-agent service is running (and risk ignoring the user thinking they have more security than they do, or should we bail out with a nice user message that this isn't a valid scenario?

I think I'm leaning towards keeping the check of just `aa_kbc_params` isn't empty before running the `init_attestation_agent`, but then within `init_attestation_agent` doing an up-front check for the binary and erroring in a nice way if it's not in the guest.

I'd love to hear others thoughts though as I might be overthinking this!

Reply:
>Yes - skopeo support for decrypting images relies on the attestation-agent to get the key, but we want to remove skopeo soon, so that shouldn't be a problem.

Thanks @stevenhorsman for explain the case!

>Do you think it's fair for us to proceed with this PR as is at the moment, or do you think we should make that kata-agent change and then think about defaulting to offline_fs_kbc in the agent-config.toml?

Yeah, I think it's perfectly OK to proceed with this PR and change agent-config.toml later. Thanks!